health.go

```diff
@@ -229,3 +244,22 @@ func (h *health) updateResult(
 	h.results[name] = result
 	return result
 }
+
+func (h *health) registerShutdownHook(cancel context.CancelFunc) {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		for range sigChan {
+			cancel()
+			return
+		}
+	}()
+}
+
+func getContext(parent context.Context, t time.Duration) (context.Context, context.CancelFunc) {
+	if t == 0 {
```

Comment:
What happens if `t < 0`?

Reply:
Then it means `t` was explicitly provided, and we use it.

Reply:
I agree that if t < 0 we should treat it as no timeout as well.